prov/verbs/configure.m4

```diff
@@ -65,6 +65,12 @@ AC_DEFUN([FI_VERBS_CONFIGURE],[
 			   [Experimental verbs features support])],
 		[])
 
+	# If the rdma-core/ibverbs libraries exist in a non-standard location, specify the
+	# runtime library path so that dependent applications and libraries can find ibverbs/rdma-core
+	AS_IF([test x"$verbs_PREFIX" != x""],
+		[verbs_ibverbs_LDFLAGS="$verbs_ibverbs_LDFLAGS -Wl,-rpath -Wl,$verbs_LIBDIR"],
```

Comment:
I'm fine with that. I've been looking into possible solutions but I'm not as familiar with runpath. As far as I know, --enable-new-dtags is needed -- as well as some other magic. Do you already know those magic invocations? (So that I might follow from what you know)

Reply:
For item 2, I know the compiler complains about the same -rpath being repeated, which is why I only did the line for libibverbs (due to common install prefix). Is there a similar concern for a runpath approach ?

Reply:
Actually... If LD_LIBRARY_PATH is preferred to avoid rpath/runpath then I'd be fine with that as well. I was just attempting to avoid any issues with adding environment variables to the travis builds. However, based on my reading -- there might be a good reason to avoid both rpath and runpath for the binaries we compile outside of travis.

Reply:
It might be better to have a configure test:

1. Does the linker accept runpath CLI params
1. If not, does it accept rpath CLI params

And down near the bottom of configure.ac:

1. Run through `LDFLAGS` and unique-ize `-L` flags
1. If runpath/rpath flags were set, then runpath-ize/rpath-ize all the `-L`'s left in `LDFLAGS`

That's essentially what Open MPI does.

Reply:
I've pulled the runpath work for the moment and here's why. RPATH and RUNPATH paths could be preferred over LD_LIBRARY_PATH for the setuid/setgid case. Now, I don't know that we;ve got a case where this would be a concern, but I'd rather not have to try to debug an environment variable issue with the libfabric binaries. 

Instead, I've modified the travis CI file to use LD_LIBRARY_PATH only and here's my rationale. As I was thinking about RPATH and RUNPATH, I was considering the case of non-ubiquitous runtime environments. I might compile libfabric on one node, and run it on another where the runtime environment is slightly different. In that specific case, I would want clear behavior with library resolution. Specifically, If I run a setuid application, I would want to fail if I could not find the expected library I wanted to use via LD_LIBRARY_PATH, rather than unexpectedly succeed because the runpath was provided and a different library was found. 

So my example above is not a common case, and if runpath/rpath is the preferred approach -- then I can pull that commit back in.

Reply:
If we don't want to cross the rpath/runpath bridge right now, I think that's fine.  My only comment was basically: if we're going to do rpath/runpath, let's do it "right".  Otherwise, let's not do it.

Meaning: using `LD_LIBRARY_PATH` -- especially for this Travis stuff -- is totally fine.

Reply:
After going deeper into the guts of linking behavior, I'd want to avoid rpath at this point. Thanks for the feedback!

Reply:
Every time I think I understand linkers, I find out that I don't know jack about linkers.